src/Fighter/Util/Binder.php

```diff
@@ -0,0 +1,33 @@
+<?hh //strict
+
+namespace Fighter\Util;
+
+trait Binder {
+    protected Map<string, mixed> $core_binds = Map {};
+
+    public function bind(string $name, mixed $bind): void {
+        if (!is_callable($bind)) {
+            throw new \InvalidArgumentException("$name is not callable");
+        }
+        $this->core_binds[$name] = $bind;
+    }
+
+    public function __call(string $name, array<mixed> $params): mixed {
+        $callback = $this->core_binds->get($name);
+
+        if($callback === null ) {
+            $last_call = current(debug_backtrace());
+            trigger_error(
```

Comment:
I tried to mimic the same error that php triggers if a method is not defined in a class

    <?php
    Class Foo{ } ;
    $foo = new Foo();
    $foo->bar();

When I run it:

    php /tmp/foo.php

    Fatal error: Call to undefined method Foo::bar() in /tmp/foo.php on line 4

The other option would be that throw an exception.